src/core/device.c

```diff
@@ -132,6 +294,23 @@ static int device_load(Unit *u) {
                         log_unit_debug_errno(u, r, "Failed to unescape name: %m");
         }
 
+        if (!verify_io_cost_supported(d)) {
+                log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "IO cost is not supported");
+                return 0;
+        }
+
+        root_slice_unit = manager_get_unit(u->manager, SPECIAL_ROOT_SLICE);
+        if (!root_slice_unit)
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Root slice does not exist");
```

Comment:
Should this be at debug level? I'm not very familiar with cgroups so I don't know if that's something that we expect to occur. If we don't, I'd log at error level here.